src/Data/Quiver/Functor.hs

```diff
@@ -98,29 +104,52 @@ class QFunctor c => QFoldable c where
   {- | Map each element of the structure to a `Monoid`,
   and combine the results.-}
   qtoMonoid :: Monoid m => (forall x y. p x y -> m) -> c p x y -> m
-  qtoMonoid f = getKQ . qfoldMap (KQ . f)
+  qtoMonoid f = getKQ #. qfoldMap (KQ #. f)
   {- | Map each element of the structure, and combine the results in a list.-}
   qtoList :: (forall x y. p x y -> a) -> c p x y -> [a]
   qtoList f = qtoMonoid (pure . f)
   {- | Map each element of a structure to an `Applicative` on a `Category`,
   evaluate from left to right, and combine the results.-}
   qtraverse_
-    :: (Applicative m, Category q)
-    => (forall x y. p x y -> m (q x y)) -> c p x y -> m (q x y)
-  qtraverse_ f = getApQ . qfoldMap (ApQ . f)
+    :: (IxApplicative m, Category q)
+    => (forall x y. p x y -> m x y (q x y)) -> c p x y -> m x y (q x y)
+  qtraverse_ f = getIxApQ #. qfoldMap (IxApQ #. f)
 instance QFoldable (ProductQ p) where qfoldMap f (ProductQ _ q) = f q
-instance QFoldable IQ where qfoldMap f (IQ c) = f c
+instance QFoldable IQ where qfoldMap f = f .# getIQ
+
+qfoldMapDefault
+  :: (QTraversable t, Category d)
+  => (forall x y . c x y -> d x y) -> t c p q -> d p q
+qfoldMapDefault f = getIxConst #. qtraverse (IxConst #. f)
 
 {- | Generalizing `Traversable` to quivers.-}
-class QFoldable c => QTraversable c where
-  {- | Map each element of a structure to an `Applicative` on a quiver,
+class (QFunctor t, QFoldable t) => QTraversable t where
+  {- | Map each element of a structure to an `IxApplicative` on a quiver,
   evaluate from left to right, and collect the results.-}
   qtraverse
-    :: Applicative m
-    => (forall x y. p x y -> m (q x y)) -> c p x y -> m (c q x y)
+    :: IxApplicative m
+    => (forall x y. c x y -> m x y (d x y))
+    -> t c p q -> m p q (t d p q)
+
 instance QTraversable (ProductQ p) where
-  qtraverse f (ProductQ p q) = ProductQ p <$> f q
-instance QTraversable IQ where qtraverse f (IQ c) = IQ <$> f c
+  qtraverse f (ProductQ p q) = ProductQ p `imap` f q
+instance QTraversable IQ where qtraverse f (IQ c) = IQ `imap` f c
+
+newtype QReverse t c x y = QReverse {getQReverse :: t (OpQ c) y x}
```

Comment:
This looks isomorphic to `ApQ t c x y` since `(OpQ c) y x` is isomorphic to `c x y`, so seems redundant

Reply:
Redundant how? The basic idea is that it's possible to implement, say, a queue that enqueues on the left and dequeues on the right by wrapping the more usual version in `QReverse`. How do you propose to do it?

Reply:
Okay, I guess it can always be done using `IxBackwards` directly. I just stole the idea from `Data.Functor.Reverse`. I guess it's at least a useful demonstration of `IxBackwards`?

Reply:
> Redundant how?
By being isomorphic to another type provided by the library

> How do you propose to do it?
I haven't given any thought really to how to port queues but presumably whatever you need the isomorphic datatype for, you can use the other one.